### PR TITLE
Normalize tests

### PR DESCRIPTION
**Pull request by UziTech**
**Marked version:** master

## Description

Normalize tests. Put non-spec tests into /new files.

New tests should be put into their own files so we don't have merge conflicts every time a PR with tests is merged.

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

**Comment by UziTech** [comment]
the tests will pass once #1439 is merged

File: test/helpers/helpers.js [hunk 1]
```diff
@@ -1,5 +1,42 @@
-var marked = require('../../lib/marked.js');
+const marked = require('../../');
+const HtmlDiffer = require('@markedjs/html-differ').HtmlDiffer;
```
Comment (styfle) [inline test/helpers/helpers.js]:
```suggestion
const { HtmlDiffer } = require('@markedjs/html-differ');
```

Might as well embrace ES6 😄

File: test/specs/run-spec.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const json = require(file);
+  const specs = json.reduce((obj, spec) => {
+    if (!obj[spec.section]) {
+      obj[spec.section] = [];
+    }
+    obj[spec.section].push(spec);
+    return obj;
+  }, {});
+
+  describe(title, function() {
```
Comment (styfle) [inline test/specs/run-spec.js]:
```suggestion
  describe(title, () => {
```

File: test/specs/run-spec.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    Object.keys(specs).forEach(section => {
+      describe(section, function() {
```
Comment (styfle) [inline test/specs/run-spec.js]:
```suggestion
      describe(section, () => {
```

File: test/specs/run-spec.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        specs[section].forEach(function(spec) {
```
Comment (styfle) [inline test/specs/run-spec.js]:
```suggestion
        specs[section].forEach(spec => {
```

File: test/specs/run-spec.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          if (options) {
+            spec.options = Object.assign({}, options, (spec.options || {}));
+          }
+          (spec.only ? fit : it)('should ' + (spec.shouldFail ? 'fail' : 'pass') + ' example ' + spec.example, function() {
```
Comment (styfle) [inline test/specs/run-spec.js]:
```suggestion
          (spec.only ? fit : it)('should ' + (spec.shouldFail ? 'fail' : 'pass') + ' example ' + spec.example, () => {
```

File: test/helpers/helpers.js [hunk 6]
```diff
@@ -1,5 +1,42 @@
...
+
+            for (var i = 0; i < expectedHtml.length; i++) {
+              if (actualHtml[i] !== expectedHtml[i]) {
+                actualHtml = actualHtml.substring(
+                  Math.max(i - 30, 0),
+                  Math.min(i + 30, actualHtml.length));
+
+                expectedHtml = expectedHtml.substring(
+                  Math.max(i - 30, 0),
+                  Math.min(i + 30, expectedHtml.length));
```
Comment (styfle) [inline test/helpers/helpers.js]: Let's move the `30` into a variable up top so this can be configurable.

File: test/helpers/helpers.js (hunk 1 above)
Comment (UziTech) [reply test/helpers/helpers.js]: That doesn't work in node 4

**Pull request merged**